libmproxy/models/connections.py

```diff
@@ -26,6 +27,7 @@ def __init__(self, client_connection, address, server):
         self.timestamp_end = None
         self.timestamp_ssl_setup = None
         self.protocol = None
+        self.socket_lock = threading.Lock()
```

Comment:
As you said, we probably don't want to touch ServerConnection at all. 
I think we can just use a wrapper for the HTTP2 protocol which does the locking?
```python
class ThreadsafeServerConnection(object):
    def __init__(self, socket_lock, ctx):
        self.socket_lock = socket_lock
        self.ctx = ctx

    def __getattr__(self, item):
        return getattr(self.ctx.server_conn, item)
    
    def send(self, message):
        with self.socket_lock:
            self.ctx.server_conn.send(msg)
```